grafast/grafserv/src/servers/hono/v4/index.ts

```diff
@@ -0,0 +1,254 @@
+import type { Context as HonoContext, Hono, MiddlewareHandler } from "hono";
+import type { StatusCode } from "hono/utils/http-status";
+
+import {
+  convertHandlerResultToResult,
+  GrafservBase,
+  makeGraphQLWSConfig,
+  normalizeRequest,
+  processHeaders,
+} from "../../../index.js";
+import type {
+  EventStreamHeandlerResult,
+  GrafservBodyJSON,
+  GrafservConfig,
+  RequestDigest,
+  Result,
+} from "../../../interfaces.js";
+import { GRAPHQL_TRANSPORT_WS_PROTOCOL, makeServer } from "graphql-ws";
+import { UpgradeWebSocket } from "hono/ws";
+import { web } from "webpack";
+
+declare global {
+  namespace Grafast {
+    interface RequestContext {
+      Hono: {
+        honoContext: HonoContext;
+      };
+    }
+  }
+}
+
+function getDigest(honoContext: HonoContext): RequestDigest {
+  const req = honoContext.req;
+  const res = honoContext.res;
+  return {
+    httpVersionMajor: 1, // Hono uses Fetch API, which doesn't expose HTTP version
+    httpVersionMinor: 1,
+    isSecure: req.url.startsWith("https"),
+    method: req.method,
+    path: req.path,
+    headers: processHeaders(req.header()),
+    getQueryParams() {
+      return req.query();
+    },
+    async getBody() {
+      const json = await req.json();
+      if (!json) {
+        throw new Error("Failed to retrieve body from hono");
+      }
+      return {
+        type: "json",
+        json,
+      } as GrafservBodyJSON;
+    },
+    requestContext: {
+      hono: {
```

Comment:
The case you had here didn't match what you added in the types (`Hono`)?

```suggestion
      honov4: {
```